test/unit/lodash.spec.js

```diff
@@ -0,0 +1,1326 @@
+                expect(elem1).to.have.deep.property('bike.saddle', 1);
+            });
+            it('should check filter string', function(){
+                expect(elem2).to.have.length(2);
+            });
+            it('should check returned value', function(){
+                expect(_.hasIn.apply(_, params)).to.eql(true);
+            });
+        });
+    });
+    
+    describe('invert', function () {
+
+        var params;
+        before(function(){
+            params = datasets.invert();
+        });
+        it('should returns inverted object', function(){
+            var obj1 = {};
+
+            //function _.times Returns the array. In this case 100 results.
+            _.times(100, function(index){
+
+                //function _.random Returns the random number between 0 and 20
+                obj1['key' + index] = _.random(20);
+
+            });
+
+            var expected = _.invert(obj1);
+            expect(_.invert.apply(_, datasets.invert(obj1))).to.eql(expected);
+        });
+    });
+
+    describe('invertBy', function () {
+        describe('invertBy1', function () {
+            var params;
+            before(function(){
+                params = datasets.invertBy1();
+            });
+            it('should return inverted object', function(){
+
+                var obj1 = {};
+
+                //function _.times Returns the array. In this case 100 results.
+                _.times(100, function(index){
+                    var indexInside = 0,
+                        counter = 0;
+                    //function _.random Returns the random number between 0 and 20
+                    obj1['key' + indexInside] = _.random(20);
+
+                    if(counter === 33 || counter === 66){
+                        indexInside++;
+                    }
+                    counter++;
+                });
+
+                var expected = _.invertBy(obj1);
+                expect(_.invertBy.apply(_, datasets.invertBy1(obj1))).to.eql(expected);
+
+            });
+        });
+        describe('invertBy2', function () {
+            var params;
+            before(function(){
+                params = datasets.invertBy2();
+            });
+            it('should return inverted object', function(){
+
+                var obj1 = {};
+
+                //function _.times Returns the array. In this case 100 results.
+                _.times(100, function(index){
+                    var indexInside = 0,
+                        counter = 0;
+                    //function _.random Returns the random number between 0 and 20
+                    obj1['key' + indexInside] = _.random(20);
+
+                    if(counter%10 === 0){
+                        indexInside++;
+                    }
+                    counter++;
+                });
+
+                var expected = _.invertBy(obj1, params[1]);
+                expect(_.invertBy.apply(_, datasets.invertBy2(obj1))).to.eql(expected);
```

Comment:
I would also assert that inverted object have property with some prefix like in current datasets is "chosen"